### ISSUE TITLE
During shrink action, do not wait for other shards

### ISSUE THREAD

**Issue by dandrestor** [elastic/curator#1088]
During the execution of a shrink action, I identified two stages at which curator will wait for shard recovery to complete:
* Moving shards to shrink node: xxx
* Applying post-shrink allocation rule ...
I suspect at these stages, curator does a GET _cluster/health?wait_for_no_relocating_shards or something similar. This makes curator to wait even for other recoveries, for indexes other than the index currently being shrunk. Curator should take a closer look at recovery and only wait for the current shards to recover.

**Comment by dandrestor** [comment]
An easy workaround for this is setting `cluster.routing.rebalance.enable` to `none` before the shrink action.

**Comment by dandrestor** [comment]
Yeah, using `cluster_routing` is definitely a good idea. But I still think that this is a workaround, and that it's worth enhancing curator so that it does not wait for _all_ recoveries, but rather just the ones that are relevant.

**Comment by untergeek** [comment]
It's dangerous to presume that it's wise or safe to worry about _only_ specific relocations.  Most clusters have limits to how many relocations can happen at a time.  As much as I may agree that this would be a beneficial feature, it's also one that is terribly easy to get wrong.  For now, I'm going to continue to expect users to handle the rebalance settings on their own.

**Comment by dandrestor** [comment]
I understand. I guess you should close this, then. 

Just for my own understanding, though, what are the problems that might occur? We are talking about two possible situations:
* Trigger a shrink operation, possibly while recoveries for other indexes are ongoing
* Trigger shard relocations, subject to cluster-wide relocation limits, possibly while other recoveries are ongoing.
What would be the problems with either of those?

**Comment by dandrestor** [comment]
I'm not talking about automatically disabling shard rebalancing. I only mentioned that as a workaround to what I see as the underlying problem, and I agree that this is something that the user should explicitly perform.

However, if the user does not disable rebalancing, and still calls the shrink operation, that is an explicit indication that the user wants the two to occur simultaneously. This is the situation in which I mean that the shrink operation should not wait for other, unrelated indexes to recover before executing. Also, the `shrink` command will still be subject to the limits regarding per-node and cluster-wide simultaneous recoveries, so the user's safeguards regarding I/O and network bandwidth will still be observed.

**Comment by untergeek** [comment]
> However, if the user does not disable rebalancing, and still calls the shrink operation, that is an explicit indication that the user wants the two to occur simultaneously.

I'm not sure I follow the logic behind this assumption.  I understand that you presume that, but why would every user?

**Comment by dandrestor** [comment]
What I mean is, the two operations (`shrink` and disabling rebalancings via `cluster_routing`) are orthogonal. The user is free to choose one or both. If the user chooses just the former, I believe `shrink` should be as atomic as possible and not force the latter upon the user.

**Comment by untergeek** [comment]
Is this fixed by the changes in #1129 ?
